### DISCUSSION TITLE
Code First Approach

### DISCUSSION THREAD

**Discussion by bsamartins** [Netflix/dgs-framework#980] [General]
Is there support for a code first approach?

**Comment by paulbakker** [comment]
Although there is [support](https://netflix.github.io/dgs/advanced/dynamic-schemas/) for creating schemas from code, the general answer is no. 

We think schema first development is the better approach and the framework is built on that concept.

**Reply by hexdecimal16** [reply]
What are adavantages of schema first approach here?

**Comment by hexdecimal16** [comment]
I believe code first approach is better (personal preference) it just avoids 
1. First writing schema 
2. Then implementing it's definations (Query/Mutation & types)

Code first approach generates schema based on code which can be further investigated and used. This way a lot of redundant steps are skipped.

**Comment by gadamsfd** [comment]
@paulbakker Using a code-first approach appears to be working at least in some cases (we've completed a functional POC using DGS with a code-first approach and haven't run into problems yet). Can you confirm that this is not supported and clarify what kind of problems could result from using a code-first approach, where we hand-code graphql type classes, potentially with additional fields and mismatched data types?
